analytics/src/main/java/com/segment/analytics/Analytics.java

```diff
@@ -1398,11 +1413,20 @@ public Analytics build() {
       factories.add(SegmentIntegration.FACTORY);
       factories.addAll(this.factories);
 
+      // Check for edge functions, disable the destination and source middleware if found
       List<Middleware> srcMiddleware = Utils.immutableCopyOf(this.sourceMiddleware);
       Map<String, List<Middleware>> destMiddleware =
           isNullOrEmpty(this.destinationMiddleware)
               ? Collections.<String, List<Middleware>>emptyMap()
               : immutableCopyOf(this.destinationMiddleware);
+      if (this.edgeFunctionMiddleware != null
+          && this.edgeFunctionMiddleware.sourceMiddleware != null) {
+        srcMiddleware = this.edgeFunctionMiddleware.sourceMiddleware;
+      }
+      if (this.edgeFunctionMiddleware != null
+          && this.edgeFunctionMiddleware.destinationMiddleware != null) {
+        destMiddleware = this.edgeFunctionMiddleware.destinationMiddleware;
+      }
```

Comment:
I think this could raise a scenario where edge function is inter-operating with either native source/destination middleware. 
Lets try to have parity with iOS here